transports/onion/src/address.rs

```diff
@@ -0,0 +1,80 @@
+use std::net::{IpAddr, SocketAddr};
+
+use arti_client::{IntoTorAddr, TorAddr, TorAddrError};
+use libp2p_core::{multiaddr::Protocol, Multiaddr};
+
+fn try_extract_socket_addr(multiaddr: &Multiaddr) -> Result<SocketAddr, TorAddrError> {
```

Comment:
Can we order the functions in this file according to their visibility / usage please?

Public functions first, followed by utilities in the order they are used as you read code from top to bottom. Thanks!